content/docs/fundamentals/hmr.md

```diff
@@ -0,0 +1,126 @@
+# Hot Module Reloading
+
+Hot Module Reloading (HMR) is a feature baked in since AdonisJS 6.x.x that allows automatic reloading of modified files without having to restart the entire server.
+
+Until recently, the traditional way of working with AdonisJS was to launch the dev server with the command `node ace serve --watch`, which would take care of completely restarting the server/node process each time a file was modified, just like Nodemon does for example.
+
+This process works well, but depending on certain things, it's sometimes not the most optimal:
+
+- Sometimes you have a lot of configured packages, and therefore providers that execute on each restart. This can slow down the startup process.
+- If you use TSX as a template engine, then a simple modification of, let's say, paddings or margins in one of your views forces you to reload the entire server, resulting in a quite slow feedback loop.
```

Comment:
This is an example linked to using a utility-first CSS framework. I don't believe this example should be shared here since it relates to other technology and not directly in TSX. We should explain that TSX views are JavaScript modules and therefore, are cached the same way as a JavaScript file.